### ISSUE TITLE
pip and wheels causing trouble

### ISSUE THREAD

**Issue by schaefi** [OSInside/kiwi#1451]
## Problem description

If you look at https://stackoverflow.com/a/46087545 it turns out recent versions of
pip (v >= 9.0.1) will try to compile a wheel even from a source distribution. This is a problem
for the way kiwi uses setup.py. In kiwi setup.py contains instructions to run code that e.g
installs man pages or the bash completion. If pip turns data into a static zip file (nothing else is a wheel) all dynamics are gone. Meaning the wheel gets created implicitly in the background and used. Using a wheel means it simply unpacks data but does not run any code. This means
no call of ```setup.py install``` will be issued. This leads to files that are expected to be installed into the system to be unpacked in the respective ```site-packages/``` directory which is wrong in the context of kiwi.

To avoid this people has to install kiwi with:

```
pip install kiwi-ng --no-binary :all:
```

Nobody will do this or remember that it is needed.

The following files are affected by the way pip works when installing kiwi:

* man pages: /usr/share/man/man8/...
* command completion: /etc/bash_completion.d/kiwi-ng.sh
* kiwi default config file: /etc/kiwi.yml
* package docs: /usr/share/doc/packages/kiwi-ng/...
     (kiwi.pdf, LICENSE, README)

kiwi stays functional without those but we should cleanup and come to a conclusion how we want to refactor setup.py in a way it seems to expected by the python community

**Comment by tomschr** [comment]
If you apply the following patch, wheel can be built (however, you need to build the manpages first):


```diff
diff --git i/setup.py w/setup.py
index 50a1b2136..eb370c4e1 100755
--- i/setup.py
+++ w/setup.py
@@ -3,6 +3,7 @@
 
 from setuptools import setup
 from setuptools.command import sdist as setuptools_sdist
+from setuptools.command import install as setuptools_install
 
 from distutils.command import build as distutils_build
 from distutils.command import install as distutils_install
@@ -122,7 +123,7 @@ class clean(distutils_clean.clean):
         distutils_clean.clean.run(self)
 
 
-class install(distutils_install.install):
+class install(setuptools_install.install):
     """
     Custom install command
     Host requirements: make
```

**Comment by schaefi** [comment]
Maybe also worth a look how other projects does it:

https://pyscaffold.org/
https://pypi.org/project/setuptools-scm/

They use setup.cfg to describe the project

**Comment by schaefi** [comment]
Fixed by #1502
